### PR TITLE
Event details role based render routing

### PR DESCRIPTION
**Pull request by thaigillespie**
I made minor changes to EventDetails and EventCard.

For EventDetails, I just removed theme.spacing() since we don't need that anymore

For EventCard, when user clicks on "See More", it now goes to the EventDetailsPage with the id of the event they clicked on in Calendar.

### CODE REVIEW

File: src/HOCs/RoutingByContextAuth.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const RoutingByContextAuth = allowedRoles => WrappedComponent => props => {
     <AuthUserContext.Consumer>
       {userClaims => {
         if (userClaims != null) {
-          for (let i = 0; i < userClaims.length; i++) {
+          for (let i = 0; i < userClaims.length; i += 1) {
```
Comment (godwinpang) [inline src/HOCs/RoutingByContextAuth.js:13]: This is the same right?

File: src/components/EventDetails/index.js [hunk 2]
```diff
@@ -117,10 +117,12 @@ EventDetailsComponent.propTypes = {
     startDate: PropTypes.shape({
       nanoseconds: PropTypes.number.isRequired,
```
Comment (godwinpang) [inline src/components/EventDetails/index.js:118]: I skimmed through this in the previous PR, just wondering why we need nanoseconds?

File: src/HOCs/RoutingByContextAuth.js (hunk 1 above)
Comment (thaigillespie) [reply src/HOCs/RoutingByContextAuth.js:13]: Yeah it's the same, it's just that the linter made me change from i++ to i += 1 lol

File: src/components/EventDetails/index.js (hunk 2 above)
Comment (thaigillespie) [reply src/components/EventDetails/index.js:118]: Oh is that not how we do it for type validation? startDate comes in as an object with nanoseconds, seconds, and toDate so I thought that was what I had to do. Do we only need to check for type for something if we use them?

Comment (godwinpang) [reply src/components/EventDetails/index.js:118]: you can use instanceOf :)

Comment (godwinpang) [reply src/components/EventDetails/index.js:118]: Proptypes is starting to wear me out a lil, maybe we can see how people feel about using something like typescript during summer lol

**Pull request merged**